### PR TITLE
Improve gzscore

### PR DESCRIPTION
**Pull request by qbarthelemy**
This PR solves issue described by https://github.com/scipy/scipy/issues/14816, with solution still not merged in SciPy https://github.com/scipy/scipy/pull/14817.

`gzscore` can be simpy re-written as the zscore of log-values. This implementation is fast, and it inherits from all options available in `zscore`: `axis`, `ddof` and `nan_policy`.

I don't know what is the deprecation policy in pingouin, but since `gzscore` should be available in SciPy 1.8.0, we could add a deprecation warning, to remove it when you will update scipy version in requirements.

### CODE REVIEW

**Comment by raphaelvallat** [comment]
Salut @qbarthelemy :-)

Thanks for the PR. I was aware of the SciPy PR and indeed wanted to deprecate the function when we'll upgrade the requirements to SciPy 1.80 (i.e. anywhere between 2-6 months after the stable SciPy release I think). In the meantime, I think your proposed changes are great, so ok to merge with me. If you want, feel free to add the deprecation warning and update the changelog with the PR. Thanks!

**Comment by qbarthelemy** [comment]
Yes, I am going to complete the PR.

However, before merging this branch in pingouin, we should probably wait for @tupui to merge the gzscore branch into SciPy.

**Comment by tupui** [comment]
> Yes, I am going to complete the PR.
> 
> 
> 
> However, before merging this branch in pingouin, we should probably wait for @tupui to merge the gzscore branch into SciPy.

Sorry this felt through the cracks. I will have a look and merge.

**Review by raphaelvallat — APPROVED** [review]

**Comment by raphaelvallat** [comment]
Thank you both! The PR looks great. However, we do get an error when running the tests @qbarthelemy:

`AttributeError: module 'warnings' has no attribute 'DeprecationWarning'`

**Comment by raphaelvallat** [comment]
Seems like we're all good to merge! Lmk when you think it'll be a good time to merge / release this. I don't anticipate there will be a new stable release of Pingouin before Jan-Feb 2022.

Merci !
Raphael

**Comment by qbarthelemy** [comment]
gzscore is already available in the dev version of scipy: http://scipy.github.io/devdocs/reference/generated/scipy.stats.gzscore.html

But, maybe we could wait the next scipy release to merge this branch.

**Comment by qbarthelemy** [comment]
I think you can merge, because the new release scipy 1.7.3 is out from 5 days! @raphaelvallat

**Pull request merged**
